### PR TITLE
Remove some deprecations and fix compile warnings

### PR DESCRIPTION
**Pull request by lenemter**
ModuleFlags.BIND_LOCAL was deprecated in 0.46, so I set required Vala version to 0.46 too.

### CODE REVIEW

File: src/AccentColor/AccentColorManager.vala [hunk 1]
```diff
@@ -151,7 +151,12 @@ public class Gala.AccentColorManager : Object {
             return null;
         }
 
-        return metadata.get_tag_string (TAG_ACCENT_COLOR);
+        try {
+            return metadata.try_get_tag_string (TAG_ACCENT_COLOR);
+        } catch (Error e) {
+            warning (e.message);
+            return null;
```
Comment (danirabbit) [inline src/AccentColor/AccentColorManager.vala:158]: Instead of return null here should we return a fallback color?

Comment (lenemter) [reply src/AccentColor/AccentColorManager.vala:158]: No, this function is used when getting accent color from wallpaper, it tries to get it from metadata and if it fails (returns null) update_accent_color selects accent color based on a picture itself.

**Comment by lenemter** [comment]
I removed deprecated `get_pkgconfig_variable` and `meson.source_root` from meson files.

**Review by danirabbit — APPROVED** [review]
Let's go :tada:

**Pull request merged**